src/gui/FileBrowser.cpp

```diff
@@ -391,12 +364,16 @@ void FileBrowser::addItems(const QString & path )
 	QDir cdir(path);
 	if (!cdir.isReadable()) { return; }
 	QFileInfoList entries = cdir.entryInfoList(
-		m_filter.split(' '), dirFilters(), QDir::LocaleAware | QDir::DirsFirst | QDir::Name | QDir::IgnoreCase);
+		m_filter.split(' '),
+		dirFilters(),
+		QDir::LocaleAware | QDir::DirsFirst | QDir::Name | QDir::IgnoreCase
+		);
 	for (const auto& entry : entries)
 	{
 		if (FileBrowser::excludedPaths().contains(entry.absoluteFilePath())) { continue; }
 
 		QString fileName = entry.fileName();
+		if (fileName.startsWith(".") && m_showHiddenContent && !m_showHiddenContent->isChecked()) continue;
```

Comment:
`fileName.startsWith(".")` is a *nix-specific way of hiding files: some systems ([more than just Windows!](https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory)) have other approaches. You can use `entry.isHidden()` for a cross-platform approach.